src/Documents/Document.ts

```diff
@@ -0,0 +1,15 @@
+import {IDocument} from './IDocument';
+
+export class Document implements IDocument {
+  public entityName(): string {
+      return this.constructor.name;
+  }
+
```

Comment:
Would we need serialize / deserialize? We are working on JS, after all. We can just do `JSON.serialize` and vice versa and get the same behavior, no?